README.md

```diff
@@ -275,6 +275,126 @@ An example policy would look like;
+            {
+                OrderId = orderId;
+            }
+            public int OrderId { get; private set; }
+        }
+````
+
+And suppose I create a message object from that class:
+
+````c#
+        var orderAccepted = new OrderAccepted(1234)
+        {
+            RaisingComponent = "my publisher",
+        };
+````
+
+This will result in the following JSON being created and populated on that Message parameter to the SNS Publish API:
+
+````json
+{
+    "OrderId": 1234,
+    "Id": "e3f84a55-b677-43df-8c24-92c170fdd89f",
+    "TimeStamp": "2019-07-03T09:53:09.2956149Z",
+    "RaisingComponent": "my sns publisher"
+}
+````
+
+#### MessageAttributes
+Any message attributes that you have chosen to send along with the message.
+
+### Sending to SQS ("point to point")
+
+JustSaying uses the SQS SendMessage API to send messages to SQS (documented [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html)). The following parameters are set:
+
+#### QueueUrl
+The URL of the queue to which the message is being sent.
+
+#### DelaySeconds
+JustSaying will set this value to be that of the `Delay` property on the `PublishMetadata` class which is supplied as a parameter to the `PublishAsync` method.
+
+#### MessageBody
+Aa JSON object containing two properties, `Subject` and `Message`. `Subject`, as with the SNS publisher, will be the type of the .NET message class. `Message` will be the actual message object, serialised to JSON. The same serialisation applies as for the SNS publisher.
+
+As an example, if I create the same OrderAccepted object as before and send that to SQS, then the value of this parameter will be:
+
+````json
+{
+    "Subject":"OrderAccepted",
+    "Message":"{\"OrderId\":1234,\"Id\":\"e3f84a55-b677-43df-8c24-92c170fdd89f\",\"TimeStamp\":\"2019-07-03T09:53:09.2956149Z\",\"RaisingComponent\":\"my sqs publisher\"}"
+}
+````
+
+This structure is to ensure that messages that are delivered to SQS queues as a result of directly publishing to the queue have the same format as messages delivered to a queue as a result of being subscribed to an SNS topic to which a message was published. So, from the point of view of a subscriber, whether the message was originally delivered to SNS or SQS is not important.
+
+### Subscribing to SQS
+
+Whether the message was originally published to an SNS topic or sent directly to an SQS queue, the subscriber will receive it by reading from an SQS queue. To do this, JustSaying uses the SQS ReceiveMessage API to receive messages from SQS (documented [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html)). The following parameters are set in JustSaying's request to this API:
+
+#### QueueUrl
+The URL of the queue from which messages are to be received.
+
+#### MaxNumberOfMessages
+JustSaying will vary this value depending on the number of CPU cores available and throttling configuration. 
```

Comment:
Call out the `Throttle` (right name?) class here?

Reply:
OK, definitely worth highlighting the point-to-point vs SNS constraint here, so that we are clear all our messages look the same whether via a dynamic recipient list router (SNS) or straight to the channel (SQS) Thought I would share some fancy terms ;-)

Reply:
Not sure I follow... Subscribing is always via SQS, whether the message originally came via SNS or SQS, so `MaxNumberOfMessages` config is independent of point-to-point / SNS.

Reply:
Added a mention of the `Throttled` class.